BasicAuthentication.md

```diff
@@ -0,0 +1,161 @@
+```swift
+public static let authenticate = ["username": "password"]
+```
+5. The function, `verifyPassword`, takes a username and password and, on success, returns a `MyBasicAuth` instance. We want to check if the password matches the user's stored password. On successful match, we initialize `MyBasicAuth` with an `id` equal to username.
+```swift
+if let storedPassword = authenticate[username], storedPassword == password {
+    callback(MyBasicAuth(id: username))
+    return
+}
+callback(nil)
+```
+This function is async, so that you can perform async actions to verify the password, e.g. looking up the username and password in a database. You must call the callback closure with either an instance of 'Self' or 'nil' before exiting 'verifyPassword'. If you do not, the server will not know to continue and you will receive a 503 "Service Unavailable" error, when you call the route.  
+
+Your complete struct should now look as follows:
+```swift
+public struct MyBasicAuth: TypeSafeHTTPBasic {
+
+    public static let authenticate = ["username": "password"]
+
+    public static func verifyPassword(username: String, password: String, callback: @escaping (MyBasicAuth?) -> Void) {
+        if let storedPassword = authenticate[username], storedPassword == password {
+            callback(MyBasicAuth(id: username))
+            return
+        }
+        callback(nil)
+    }
+
+    public var id: String
+}
+```
+
+### Adding Basic Auth to the routes
+
+`MyBasicAuth` is a Type-safe middleware and can be registered to a Codable route by adding it to the handler signature.
+
+Add `auth: MyBasicAuth` to the completion closure in the storeHandler signature.
+```swift
+func storeHandler(auth: MyBasicAuth, meal: Meal, completion: @escaping (Meal?, RequestError?) -> Void ) {
+```
+Add `auth: MyBasicAuth` to the completion closure in the loadHandler signature.
+```swift
+func loadHandler(auth: MyBasicAuth, completion: @escaping ([Meal]?, RequestError?) -> Void ) {
+```
+Add `auth: MyBasicAuth` to the completion closure in the summaryHandler signature.
+```swift
+func summaryHandler(auth: MyBasicAuth, completion: @escaping (Summary?, RequestError?) -> Void ) {
+```
+
+These routes now require basic authentication to use. You can test this by running the server and going to [http://localhost:8080/summary](http://localhost:8080/summary).  
+
+The request will be rejected as unauthorized and your browser will offer a window for you to input the username and password.  
+
+Enter "username" and "password" to be allowed to view the route or any other combination to have the request rejected.  
+
+The browser will store correct credentials so use a private window if you want to test rejected credentials.
+
+Congratulations!!! You have just added HTTP basic authentication to your Kitura server.
+
+### Sending Credentials from KituraKit
+
+You now need to send the username and password from KituraKit so that your client can still connect to your protected routes.
```

Comment:
You need to enable your food tracker mobile application to send the username and password so that it can continue to connect to your now protected routes. You can do this by setting the default credentials on the KituraKit client as shown below.